ClientLib/src/main/java/org/droidplanner/services/android/impl/core/drone/autopilot/apm/solo/ArduSolo.java

```diff
@@ -257,7 +278,11 @@ public void run() {
     @Override
     public void notifyDroneEvent(final DroneInterfaces.DroneEventsType event) {
         switch (event) {
-            case HEARTBEAT_FIRST:
+            case HEARTBEAT_FIRST: {
```

Comment:
Turns out it does. I'll also investigate why I'm getting double disconnect messages as well.